src/librustdoc/html/static/main.js

```diff
@@ -502,54 +597,92 @@
             });
 
-            $(document).off('keydown.searchnav');
-            $(document).on('keydown.searchnav', function(e) {
-                var $active = $results.filter('.highlighted');
+            var search_input = document.getElementsByClassName('search-input')[0];
+            search_input.onkeydown = null;
+            search_input.onkeydown = function(e) {
+                var actives = [];
+                onEach(document.getElementsByClassName('search-results'), function(e) {
+                    onEach(document.getElementsByClassName('highlighted'), function(e) {
+                        actives.push(e);
+                    });
+                });
 
                 if (e.which === 38) { // up
-                    if (!$active.length || !$active.prev()) {
+                    if (!actives.length || !actives[0].previousElementSibling) {
                         return;
                     }
 
-                    $active.prev().addClass('highlighted');
-                    $active.removeClass('highlighted');
+                    addClass(actives[0].previousElementSibling, 'highlighted');
+                    removeClass(actives[0], 'highlighted');
                 } else if (e.which === 40) { // down
-                    if (!$active.length) {
-                        $results.first().addClass('highlighted');
-                    } else if ($active.next().length) {
-                        $active.next().addClass('highlighted');
-                        $active.removeClass('highlighted');
+                    if (!actives.length) {
+                        var results = document.getElementsByClassName('search-results');
+                        if (results.length > 0) {
+                            var res = results[0].getElementsByClassName('result');
+                            if (res.length > 0) {
+                                addClass(res[0], 'highlighted');
+                            }
+                        }
+                    } else if (actives[0].nextElementSibling) {
+                        addClass(actives[0].nextElementSibling, 'highlighted');
+                        removeClass(actives[0], 'highlighted');
                     }
                 } else if (e.which === 13) { // return
-                    if ($active.length) {
-                        document.location.href = $active.find('a').prop('href');
+                    if (actives.length) {
+                        document.location.href = actives[0].getElementsByTagName('a')[0].href;
                     }
-                } else {
-                  $active.removeClass('highlighted');
+                } else if (actives.length > 0) {
+                    removeClass(actives[0], 'highlighted');
                 }
-            });
+            };
         }
 
         function escape(content) {
-            return $('<h1/>').text(content).html();
+            let h1 = document.createElement('h1');
+            h1.innerHTML = content;
```

Comment:
Not sure because `h1` can have html content as well (`code` balise and stuff for examples).